\documentclass{beamer}
\usepackage{tikz}
\usepackage{graphicx}
\usepackage{tabu}
\usetikzlibrary{positioning}
\usetikzlibrary{calc}
%\usetikzlibrary{arrows}

\usefonttheme{professionalfonts}
\usetheme{Boadilla}
\setbeamertemplate{navigation symbols}{}%remove navigation symbols
\hypersetup{pdfstartview={Fit}} % fits the presentation to the window when first displayed
\graphicspath{{./figures/}{./figures/generated/}{./figures/static/}}

%Edwards PB, Wanjura WJ, Brown WV: Selective herbivory by Christmas beetles in response to intraspecific variation in Eucalyptus terpenoids. Oecologia 1993, 95:551–557

%Info
\title[Eucalyptus Mutation Topology]{Patterns of mutation match physical branching topology in Eucalyptus}
\titlegraphic{
	\includegraphics[width=.3\linewidth]{gatk_tree_rightwards.pdf}
	\includegraphics[width=.3\textwidth, angle=90]{labeled_tree.jpg}
	}
\date{6/27/17}
\author{Adam Orr}

\begin{document}
\frame{\titlepage}

\begin{frame}{Why look for somatic mutations?}
	\begin{tabu} to .95\textwidth { X[c] | X[c] }
	% row 1 col 1
	\textbf{Disease} & \textbf{Development}
	\\
	Cancer \& Somatic mutation rate & Understanding relationship between tissues
	\\
	\\
	\hline
	\\
	\textbf{Agriculture} & \textbf{Testing Methods}
	% row 2 col 1
	\\
	Looking for interesting phenotypes in clonally reproducing species & Using a known pattern of relationships to evaluate method performance
	\end{tabu}
\end{frame}

\begin{frame}{Mutations in Plants}

	Little is known about the process in plants.

	\vfill

	Genome-level approach required to find enough mutations, but plant genomes are difficult to work with due to highly repetitive genomes.

	\vfill

	Relatively few plant genomes.

	% \begin{itemize}
	% 	\item Do plants have a segregated germline?
	% 	\item Do plants induce mutation in response to stress?
	% \end{itemize}

	% To answer these questions, we must first have methods to sensitively detect mutations.

\end{frame}

\begin{frame}{Mutations are very rare, but sequencing errors are very common.}

\textbf{Sequencing error} alone is \textbf{$\sim10^{-2}$} while mutation rate after error-checking is \textbf{$\sim10^{-10}$}

\vfill

Somatic mutations are hard to find.

\begin{itemize}
\item Errors accumulate during PCR prior to sequencing - then propagate
\item Errors accumulate in amplification steps during sequencing
\item Technical error from sequencer
\end{itemize}

\end{frame}

\begin{frame}{A Genetic Mosaic}
	\begin{center}
	\includegraphics[width=.6\linewidth]{unlabeled_tree.jpg}
	\end{center}
	\begin{itemize}
		\item Edwards identified as mosaic in 1993\footnote{\textit{Edwards PB, Wanjura WJ, Brown WV. Oecologia 1993, 95:551–557.}}
		\item Sheep pen in Yeoval, New South Wales
		\item Differential oil production gives protection from Christmas beetles
		% \item Is this mutation a controlled process?
	\end{itemize}
\end{frame}

\begin{frame}{Study Methodology}
\begin{columns}

\column{.5\linewidth}
\begin{itemize}
\item Sequence 8 samples in triplicate
\item $\sim$10X coverage for each replicate
\item Align sequence to genome of \textit{Eucalyptus grandis}
\item Use replicates to remove false positives
\end{itemize}

\column{.5\linewidth}
\begin{center}
\begin{tikzpicture}[sqnode/.style={rectangle, draw=black!60, fill=black!5,very thick,minimum size=1cm}]
	\node[sqnode] (sequencing) {Sequence};
	\node[sqnode] (alignment) [below = of sequencing] {Align (NextGenMap)};
	\node[sqnode] (varcall) [below = of alignment] {Get Variants (GATK)};
	\node[sqnode] (flt) [below = of varcall] {Filter};
	\draw[ultra thick,->] (sequencing.south) -- (alignment.north);
	\draw[ultra thick,->] (alignment.south) -- (varcall.north);
	\draw[ultra thick,->] (varcall.south) -- (flt.north);
\end{tikzpicture}
\end{center}
\end{columns}
\end{frame}

\begin{frame}{Mutation Pattern Approximately Matches Tree Structure}
\begin{columns}
\column{.5\linewidth}
	\begin{center}
	GATK Best Practices Tree
	\end{center}
	\includegraphics[width=\linewidth]{gatk_tree_rightwards.pdf}
\column{.5\linewidth}
	\begin{center}
	True Tree
	\end{center}
	\includegraphics[width=\linewidth,angle=90]{true_tree.pdf}
\end{columns}
\end{frame}

\begin{frame}{Most Reads Are Not Mapped to the \textit{E. grandis} Reference}
\begin{center}
\includegraphics[width=\linewidth]{bowtie1_hist.pdf}
\end{center}
\end{frame}

\begin{frame}{A Reference-Free Method Performs Similarly}
	\begin{columns}
	\column{.5\linewidth}
		\begin{center}
		DiscoSNP++ Tree
		\end{center}
		\includegraphics[width=\linewidth]{disco_tree_rightwards.pdf}
	\column{.5\linewidth}
		\begin{center}
		True Tree
		\end{center}
		\includegraphics[width=\linewidth,angle=90]{true_tree.pdf}
	\end{columns}
\end{frame}


\begin{frame}{Approximating a Genome}

Use \textit{E. melliodora} genome as a starting place, then generate a new reference and map to that reference.

\begin{center}
	\begin{tikzpicture}[cnode/.style={rectangle,draw=black!60,fill=black!5,very thick}, node distance = 1 cm]
		\node[cnode] (reads){Reads};
		\node[cnode] (eg)[below = .5cm of reads]{\textit{E. grandis}};
		\node[cnode] (em1)[right = of eg]{\textit{E. melliodora} 1};
		\node[cnode] (em2)[right = of em1]{\textit{E. melliodora} 2};
		\node[cnode] (etc)[right = of em2] {...};
		\draw[line width=.8mm,->] ($(reads.east)-(0cm,.1cm)$) -- ($(reads.east)+(.5cm,-.1cm)$) |- (em1.west);
		\draw[line width=.8mm,->] (reads.east) -- ($(reads.east)+(4.3cm,0cm)$) |- (em2.west);
		\draw[line width=.8mm,->] ($(reads.east)+(0cm,.1cm)$) -- ($(reads.east)+(8cm,.1cm)$) |- (etc.west);
		\draw[line width=.8mm,->] (eg) -- (em1);
		\draw[line width=.8mm,->] (em1) -- (em2);
		\draw[line width=.8mm,->] (em2) -- (etc);
	\end{tikzpicture}
\end{center}


\end{frame}

\begin{frame}{Our New Reference Improves Mapping Quality}
\begin{center}
\includegraphics[width=.95\linewidth]{both_hist.pdf}
\end{center}
\end{frame}

\begin{frame}{Filtering Variants}
Remove variants likely from alignment errors:
\begin{itemize}
	\item at sites with excessive depth ($>$500).
	\item with excessive levels of heterozygosity.
	\item within 50 bases of an indel.
	\item in repeat regions 
\end{itemize}
\end{frame}


\begin{frame}{Removing Variants in Repeat Regions Improves Tree Topology}
	\begin{columns}
	\column{.5\linewidth}
		\begin{center}
		Predicted Variants
		\end{center}
		\includegraphics[width=\linewidth]{gatk_repeats_removed.pdf}
	\column{.5\linewidth}
		\begin{center}
		True Tree
		\end{center}
		\includegraphics[width=\linewidth,angle=90]{true_tree.pdf}
	\end{columns}
\end{frame}

\begin{frame}{Acknowledgements}
\begin{itemize}
\item Reed Cartwright
\item Robert Lanfear, Australian National University
\end{itemize}

\begin{columns}
\column{.4\linewidth}
	\includegraphics[width=.9\linewidth]{lab_logo.pdf}
	\\~\\
	\includegraphics[width=.9\linewidth]{biodesign_logo.pdf}
\column{.4\linewidth}
	\includegraphics[width=.9\linewidth]{sols_logo.pdf}
	\\~\\
	This work is supported by grants NIH R01-HG007178 and NSF DBI-1356548.
\end{columns}

\end{frame}

\end{document}