Detection of somatic mutations in *Eucalyptus melliodora*
=========================================================

Finding variants (SNPs, insertions, and deletions) is traditionally an involved process
in which Next-Gen Sequencing reads are compared to a reference, and reads which
contain changes are reported as variants. In non-model species for which there
is no genome, this requires *de novo* assembly of a genome. This is computationally expensive
and time-consuming. New reference-free methods promise fast and accurate detection of of variants.
I will discuss one such method and evaluate its performance against a traditional pipeline,
using an individual of the species *Eucalyptus melliodora*. This individual will allow
us to evaluate phylogenetic tools and has the potential to reveal agriculturally-relevant
variation. A similar method could be used for understanding the spread of somatic mutations in mammals.